services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -395,16 +391,9 @@ protected Answer copyFromS3ToNfs(CopyCommand cmd, DataTO srcData, S3TO s3, DataT
                     return new CopyCmdAnswer(errMsg);
                 }
             }
-
             File destFile = new File(downloadDirectory, substringAfterLast(srcData.getPath(), S3Utils.SEPARATOR));
-
             S3Utils.getFile(s3, s3.getBucketName(), srcData.getPath(), destFile).waitForCompletion();
 
-
-            if (destFile == null) {
```

Comment:
I removed that because the new File (...) will always create an object that is referenced by destFile; therefore, it will never be == null.